client/src/components/Form/Form.js

```diff
@@ -1,38 +1,52 @@
-import React, { Component } from 'react';
+import React, { useEffect, useRef } from 'react';
 import FormAlert from 'components/FormAlert/FormAlert';
 import PropTypes from 'prop-types';
 
-class Form extends Component {
-  constructor(props) {
-    super(props);
-    this.handleSubmit = this.handleSubmit.bind(this);
-  }
+const Form = ({
+  autoFocus,
+  setDOM = () => null,
+  valid,
+  actions,
+  fieldHolder,
+  actionHolder = {
+    className: 'btn-toolbar'
+  },
+  afterMessages,
+  attributes,
+  fields,
+  handleSubmit,
+  mapActionsToComponents,
+  mapFieldsToComponents,
+  messages,
+  formTag: FormTag = 'form',
+  FormAlertComponent = FormAlert,
+}) => {
+  const formRef = useRef(null);
 
-  componentDidMount() {
-    if (!this.props.autoFocus) {
+  useEffect(() => {
+    if (!autoFocus) {
       return;
     }
 
-    if (this.form) {
-      const input = this.form.querySelector('input:not([type=hidden]), select, textarea');
+    if (formRef.current) {
+      const input = formRef.current.querySelector('input:not([type=hidden]), select, textarea');
       if (input) {
         input.focus();
         if (input.select) {
           input.select();
         }
       }
     }
-  }
+  }, [autoFocus]);
```

Comment:
```suggestion
  }, []);
```
This replaces `componentDidMount()` so the dependency array should be empty

Reply:
Updated